rskj-core/src/test/java/co/rsk/mine/MinerManagerTest.java

```diff
@@ -263,32 +253,29 @@ public void mineBlock() {
         Assert.assertFalse(blockchain.getBestBlock().getTransactionsList().isEmpty());
 
         SnapshotManager snapshotManager = new SnapshotManager();
-        snapshotManager.resetSnapshots(blockchain);
+        snapshotManager.resetSnapshots(blockchain, transactionPool);
 
         Assert.assertEquals(0, blockchain.getBestBlock().getNumber());
 
         manager.mineBlock(blockchain, minerClient, minerServer);
         manager.mineBlock(blockchain, minerClient, minerServer);
         Assert.assertEquals(2, blockchain.getBestBlock().getNumber());
 
-        snapshotManager.resetSnapshots(blockchain);
-        Assert.assertTrue(blockchain.getTransactionPool().getPendingTransactions().isEmpty());
+        snapshotManager.resetSnapshots(blockchain, transactionPool);
+        Assert.assertTrue(transactionPool.getPendingTransactions().isEmpty());
 
         manager.mineBlock(blockchain, minerClient, minerServer);
 
-        Assert.assertTrue(blockchain.getTransactionPool().getPendingTransactions().isEmpty());
+        Assert.assertTrue(transactionPool.getPendingTransactions().isEmpty());
```

Comment:
I think it's not on the scope of PR but why are we testing conditions on transactionPool here¿?

Reply:
It makes sense because the mining facilities interact with the transaction pool, but it doesn't seem to be well tested.